### PR TITLE
[core] Small redis context cleanup

### PR DESCRIPTION
**Pull request by vitsai**
Very quick cleanup of dead code in redis context.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/redis_context.cc [hunk 1]
```diff
@@ -323,11 +323,8 @@ void SetDisconnectCallback(RedisAsyncContext *redis_async_context) {
 }
 
 void FreeRedisContext(redisContext *context) { redisFree(context); }
-
 void FreeRedisContext(redisAsyncContext *context) {}
```
Comment (jjyao) [inline src/ray/gcs/redis_context.cc:326]: Is this used?

Comment (vitsai) [reply src/ray/gcs/redis_context.cc]: removed

File: src/ray/gcs/redis_context.cc [hunk 2]
```diff
@@ -347,12 +341,7 @@ Status ConnectWithoutRetries(const std::string &address,
     }
     return Status::RedisError(oss.str());
   }
-  if (context != nullptr) {
-    // Don't crash if the RedisContext** is null.
-    *context = newContext;
-  } else {
-    FreeRedisContext(newContext);
-  }
+  *context = newContext;
```
Comment (jjyao) [inline src/ray/gcs/redis_context.cc:344]: Add a RAY_CHECK that context is not null

**Review by jjyao — APPROVED** [review]

**Comment by jjyao** [comment]
Hmm, there are many test failures.

**Comment by vitsai** [comment]
added back the unused asioclients, apparently we need them to stay alive

**Pull request merged**